app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavoriteButton.kt

```diff
@@ -0,0 +1,41 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+private var isFavorite = false
+
+@Composable
+fun FavoriteButton(
+    backgroundColor: Color
+) {
+    Image(
+        painter = painterResource(id = if (isFavorite) R.drawable.ic_favorite_filled else R.drawable.ic_favorite),
+        contentDescription = null,
+        modifier = Modifier
+            .clickable {
+                isFavorite = isFavorite.not()
```

Comment:
here call `onClick()` instead of this line because you will do the logic in Preview. 
Try to think that you are making parts of something bigger, when you get to the Preview you are combining part or the whole design of a component. Then try to avoid some logic that is not really necessary to implement into smaller parts of your component. This gives you flexibility to reuse it on different place with different parameters or logic.